frontendUiTableDemo/src/main/java/com/virtuslab/gitmachete/frontend/ui/demo/DemoGitMacheteRepository.java

```diff
@@ -0,0 +1,227 @@
+package com.virtuslab.gitmachete.frontend.ui.demo;
+
+import java.time.Instant;
+
+import io.vavr.NotImplementedError;
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang.NotImplementedException;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.common.value.qual.ArrayLen;
+
+import com.virtuslab.branchlayout.api.IBranchLayout;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteCommit;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteForkPointCommit;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteNonRootBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRemoteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepository;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRootBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMergeParameters;
+import com.virtuslab.gitmachete.backend.api.IGitRebaseParameters;
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+
+public class DemoGitMacheteRepository implements IGitMacheteRepository {
+
+  private final List<IGitMacheteRootBranch> roots;
+
+  public DemoGitMacheteRepository() {
+    var nullPointedCommit = new Commit("");
+    var fp = new FpCommit("Fork point commit");
+    NonRoot[] nonRoots = {
+        new NonRoot(/* name */ "allow-ownership-link",
+            /* customAnnotation */ "#Gray edge: branch is merged to its parent branch",
+            nullPointedCommit,
+            /* fork point */ null,
+            /* downstreamBranches */ List.empty(),
+            /* commits */ List.empty(),
+            SyncToParentStatus.MergedToParent),
+        new NonRoot(/* name */ "build-chain",
+            /* customAnnotation */ "#Green edge: branch is in sync to its parent branch",
+            nullPointedCommit,
+            /* fork point */ null,
+            /* downstreamBranches */ List.empty(),
+            /* commits */ List.of(new Commit("Second commit of build-chain"),
+                new Commit("First commit of build-chain")),
+            SyncToParentStatus.InSync),
+        new NonRoot(/* name */ "call-ws",
+            /* customAnnotation */ "#Yellow edge: Branch is in sync to its parent branch but the fork point is not equal to parent branch",
```

Comment:
To increase readability:
```suggestion
            /* customAnnotation */ "# Yellow edge: Branch is in sync to its parent branch, but the fork point is NOT equal to parent branch",
```